infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntime.java

```diff
@@ -139,6 +130,48 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
     }
   }
 
+  /**
+   * Bootstraps machines in parallel.
+   *
+   * @throws InfrastructureException when any error occurs while bootstrapping machines
+   * @throws InterruptedException when machines bootstrapping was interrupted
+   */
+  private void bootstrapMachines() throws InfrastructureException, InterruptedException {
+    List<CompletableFuture> checkMachines = new ArrayList<>(machines.size());
+    final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
I see that using ExecutorService is maybe the easiest way to implement task interruption, but I'm not sure that it's a good idea to create new executor for each workspace start.
BTW it should have the naming format.

Reply:
So why than you suggested this in your [comment]?(https://github.com/eclipse/che/pull/7379#discussion_r151351793)

Reply:
I meant using one Executor to OpenShiftInternalRuntime class (like WorkspaceSharedPool) but not one for each workspace start. Sorry that it wasn't clear enough.